src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -0,0 +1,290 @@
+            _intervalStopwatch = ValueStopwatch.StartNew();
+            MetricsFilePath = _optionsMonitor.CurrentValue.MetricsFilePath;
+
+            IsAlwaysReady = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsAlwaysReadyInstance) == "1";
+        }
+
+        internal async Task OnPublishMetrics()
+        {
+            if (FunctionExecutionCount == 0 && FunctionExecutionTimeMS == 0 && !IsAlwaysReady)
+            {
+                // no activity to report
+                return;
+            }
+
+            // we've been accumulating function activity for the entire period
+            // publish this activity and reset
+            Metrics metrics = null;
+            lock (_lock)
+            {
+                metrics = new Metrics
+                {
+                    TotalTimeMS = (long)_intervalStopwatch.GetElapsedTime().TotalMilliseconds,
+                    ExecutionCount = FunctionExecutionCount,
+                    ExecutionTimeMS = FunctionExecutionTimeMS,
+                    IsAlwaysReady = IsAlwaysReady
+                };
+
+                FunctionExecutionTimeMS = FunctionExecutionCount = 0;
+                _instanceActivityStopwatch = ValueStopwatch.StartNew();
+            }
+
+            await PublishMetricsAsync(metrics);
+        }
+
+        private async void OnFunctionMetricsPublishTimer(object state)
+        {
+            await OnPublishMetrics();
+        }
+
+        private async Task PublishMetricsAsync(Metrics metrics)
+        {
+            try
+            {
+                bool metricsPublishEnabled = !string.IsNullOrEmpty(MetricsFilePath);
+                if (metricsPublishEnabled && !PrepareDirectoryForFile(_fileSystem))
+                {
+                    return;
+                }
+
+                string metricsContent = JsonConvert.SerializeObject(metrics);
+                _logger.PublishingMetrics(metricsContent);
+
+                if (metricsPublishEnabled)
+                {
+                    string fileName = $"{Guid.NewGuid().ToString().ToLower()}.json";
+                    string filePath = Path.Combine(MetricsFilePath, fileName);
+
+                    using (var streamWriter = _fileSystem.File.CreateText(filePath))
+                    {
+                        await streamWriter.WriteAsync(metricsContent);
+                    }
+                }
+            }
+            catch (Exception ex) when (!ex.IsFatal())
+            {
+                _logger.LogError(ex, "Error writing metrics file.");
+            }
+        }
+
+        private bool PrepareDirectoryForFile(IFileSystem fileSystem)
+        {
+            if (string.IsNullOrEmpty(MetricsFilePath))
+            {
+                return false;
+            }
+
+            // ensure the directory exists
+            _fileSystem.Directory.CreateDirectory(MetricsFilePath);
```

Comment:
Use the parameter passed to this method (`fileSystem`) ?. Same for line 161

Reply:
Removed the parameter